### PR TITLE
Cleaned Up DocumentationGeneration.cs, EncryptionServices.cs ExtentionMethods.cs Files

### PR DESCRIPTION
**Pull request by ob-ab**
- Splitted up classes from DocumentationGeneration.cs File into separate files and cleaned it up
- Cleaned up EncryptionServices.cs
- Cleaned up ExtensionMethods.cs
- Updated dependent file(s)

### CODE REVIEW

File: taskt/Core/EncryptionServices.cs [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
 {
     public static class EncryptionServices
     {
         // This size of the IV (in bytes) must = (keysize / 8).  Default keysize is 256, so the IV must be
         // 32 bytes long.  Using a 16 character string here gives us 32 bytes when converted to a byte array.
-        private const string initVector = "kPcmAhmtL4ofXSMJ";
+        private const string InitVector = "kPcmAhmtL4ofXSMJ";
         // This constant is used to determine the keysize of the encryption algorithm
-        private const int keysize = 256;
+        private const int KeySize = 256;
```
Comment (melugi) [inline taskt/Core/EncryptionServices.cs]: The fields are private can you change to `_initVector` and `_keySize`?

**Review by melugi — CHANGES_REQUESTED** [review]
Looks good, just two private variables were misnamed. Thanks a lot!

File: taskt/Core/EncryptionServices.cs (hunk 1 above)
Comment (ob-ab) [reply taskt/Core/EncryptionServices.cs]: Do you remember, we agreed upon the **(PascalCase)** naming convention for **Constants** and Enums?

Comment (openbots-ff) [reply taskt/Core/EncryptionServices.cs]: Did we? I don't remember that

File: taskt/Core/Documentation/DocumentationGeneration.cs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                    stringBuilder.AppendLine("|" + commandLabel + "|" + helpfulExplanation + "|" + sampleUsage + "|" + remarks + "|");
+                }
+
+                stringBuilder.AppendLine(Environment.NewLine);
+
+                stringBuilder.AppendLine("## Developer/Additional Reference");
+                stringBuilder.AppendLine("Automation Class Name: " + commandClass.Name);
+                stringBuilder.AppendLine("Parent Namespace: " + commandClass.Namespace);
+                stringBuilder.AppendLine("This page was generated on " + DateTime.Now.ToString("MM/dd/yy hh:mm tt"));
+
```
Comment (openbots-ff) [inline taskt/Core/Documentation/DocumentationGeneration.cs]: Minor change, please remove some of the whitespace between the stringbuilder.AppendLine()s. Both here and further down.

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: taskt/Core/EncryptionServices.cs (hunk 1 above)
Comment (ob-ab) [reply taskt/Core/EncryptionServices.cs]: Yes, we did.

File: taskt/Core/Documentation/DocumentationGeneration.cs (hunk 2 above)
Comment (ob-ab) [reply taskt/Core/Documentation/DocumentationGeneration.cs]: done

File: taskt/Core/EncryptionServices.cs (hunk 1 above)
Comment (CameronHerwig) [reply taskt/Core/EncryptionServices.cs]: This still needs to be adjusted

**Review by CameronHerwig — CHANGES_REQUESTED** [review]

File: taskt/Core/EncryptionServices.cs (hunk 1 above)
Comment (ob-ab) [reply taskt/Core/EncryptionServices.cs]: updated

**Review by CameronHerwig — DISMISSED** [review]

File: taskt/Core/Utilities/CommonUtilities/VariableMethods.cs [hunk 3]
```diff
@@ -209,14 +201,12 @@ public static string ConvertToUserVariable(this String str, object sender)
                         {
                             str = str.Replace(potentialVariable, (string)varCheck.GetDisplayValue());
                         }
-                        catch (Exception)
+                        catch (Exception ex)
                         {
-
+                            throw new Exception(ex.Message);
```
Comment (openbots-ff) [inline taskt/Core/Utilities/CommonUtilities/VariableMethods.cs]: Is there any point to this catch if it's just re-throwing the exception?

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: taskt/Core/Utilities/CommonUtilities/VariableMethods.cs (hunk 3 above)
Comment (ob-ab) [reply taskt/Core/Utilities/CommonUtilities/VariableMethods.cs]: It's just re-throwing exception.
Updated it.

**Review by openbots-ff — APPROVED** [review]

**Review by melugi — APPROVED** [review]
Looks good, thanks for the changes. Maybe we should get rid of the StringMethods file in the future. It doesn't seem to add much and half the methods act as aliases for already existing methods.

**Pull request merged**